src/main/java/core/basesyntax/MyLinkedList.java

```diff
@@ -3,45 +3,141 @@
 import java.util.List;
 
 public class MyLinkedList<T> implements MyLinkedListInterface<T> {
+    private int size;
+    private Node<T> head;
+    private Node<T> tail;
+
     @Override
     public void add(T value) {
+        Node<T> prevNode = tail;
+        Node<T> newNode = new Node<>(prevNode, value, null);
+        tail = newNode;
+        if (prevNode == null) {
+            head = newNode;
+        } else {
+            prevNode.next = newNode;
+        }
+        size++;
     }
 
     @Override
     public void add(T value, int index) {
+        if (index == size) {
+            add(value);
+        } else if (index == 0) {
+            Node<T> newNode = new Node<>(null, value, tail);
+            newNode.next = head;
+            head.prev = newNode;
+            head = newNode;
+            size++;
+        } else {
+            Node<T> prevNode = findElementByIndex(index).prev;
+            Node<T> nextNode = findElementByIndex(index);
```

Comment:
```suggestion
            Node<T> nextNode = findElementByIndex(index);
            Node<T> prevNode = nextNode.prev;
```
don't call same method 2 times